handler/notification_handler.py

```diff
@@ -32,67 +33,101 @@ def push(self, notification:Notification):
+            c.execute('SELECT * from subscription')
+            row = c.fetchone()
+            subscription_info = dict(zip(self.schema, row))
+            subscription_info = {
+                'endpoint': subscription_info['endpoint'],
+                'keys': {
+                    'p256dh': subscription_info['p256dh'],
+                    'auth': subscription_info['auth']
+                }
             }
-        }
-        conn.close()
+        except sqlite3.OperationalError as e:
+            subscription_info = None
+        finally:
+            conn.close()
         return subscription_info
 
+
     def push(self, notification:Notification):
-        try:
-            webpush(
-                subscription_info=self.get_subscription(),
-                data=json.dumps(notification),
-                vapid_private_key="sec/vapid_private_key.pem",
-                vapid_claims={
-                        "sub": "mailto:buerro@icloud.com"
-                    }
-            )
-        except WebPushException as ex:
-            print("Failed to push notification: {}", repr(ex))
-
-            if ex.response and ex.response.json():
-                extra = ex.response.json()
-                print("Remote service replied with a {}:{}, {}",
-                      extra.code,
-                      extra.errno,
-                      extra.message
+        def _push(notification:Notification):
+            key_file="sec/vapid_private_key.pem"
+            if pathlib.Path(key_file).exists():
+                priv_key = key_file
+            else:
+                # snagged from pywebpush/test
+                priv_key = (
+                    "MHcCAQEEIPeN1iAipHbt8+/KZ2NIF8NeN24jqAmnMLFZEMocY8RboAoGCCqGSM49"
+                    "AwEHoUQDQgAEEJwJZq/GN8jJbo1GGpyU70hmP2hbWAUpQFKDByKB81yldJ9GTklB"
+                    "M5xqEwuPM7VuQcyiLDhvovthPIXx+gsQRQ=="
+                )
+
+            subscription = self.get_subscription()
+            if not subscription:
+                raise Exception("No user subscribed to notifications.")
+            try:
+                print(f"pushing to {subscription['endpoint']}")
+                webpush(
+                    subscription_info=subscription,
+                    data=json.dumps(notification),
+                    vapid_private_key=priv_key,
+                    vapid_claims={
+                            "sub": "mailto:buerro@icloud.com"
+                        }
                 )
+            except WebPushException as ex:
+                print("Failed to push notification: {}", repr(ex))
+
+                if ex.response and ex.response.json():
+                    extra = ex.response.json()
+                    print("Remote service replied with a {}:{}, {}",
+                          extra.code,
+                          extra.errno,
+                          extra.message
+                    )
+        store = UsecaseStore.instance()
+        if store.get_running() is not None:
```

Comment:
Push notification only if no use cases is running. Otherwise register the push as a callback on the finish of the current use case.